thunder/src/detailed.cc

```diff
@@ -277,6 +279,15 @@ void DetailedPlacer::set_fold_reg(const ::vector<::string> &cluster_blocks,
     }
 }
 
+void DetailedPlacer::set_placer_cost_exp() {
+    int exp = 1;
+    char* dist_exp = std::getenv("PNR_PLACER_EXP");
+    if (dist_exp != NULL) {
```

Comment:
Use either

```C++
if (!dist_exp)
```

of
```C++
if (dist_exp != nullptr)
```

`NULL` is old C syntax which is not recommended in C++.

Other than this, the PR looks good to me.